core/amber/src/main/python/core/architecture/packaging/input_manager.py

```diff
@@ -191,21 +189,7 @@ def _process_marker(self, marker: Marker) -> Iterator[InternalMarker]:
             yield StartOfInputPort()
         if isinstance(marker, EndOfInputChannel):
             channel = self._channels[self._current_channel_id]
-            channel.complete()
-            port_id = channel.port_id
-            port_completed = all(
-                map(
-                    lambda channel: channel.is_completed(),
-                    self._ports[port_id].channels,
-                )
-            )
-
-            if port_completed:
-                yield EndOfInputPort()
-
-            all_ports_completed = all(
-                map(lambda port: port.is_completed(), self._ports.values())
-            )
-
-            if all_ports_completed:
+            self._ports[channel.port_id].completed = True
+            yield EndOfInputPort()
+            if all(port.completed for port in self._ports.values()):
                 yield EndOfOutputPorts()
```

Comment:
This part should be integrated into EndChannelHandler, like what you did for Scala side.

Reply:
Moved.